api/src/test/java/org/pmiops/workbench/api/DataDictionaryTest.java

```diff
@@ -0,0 +1,156 @@
+package org.pmiops.workbench.api;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.sql.Timestamp;
+import java.time.Clock;
+import java.time.Instant;
+import java.time.ZoneId;
+import javax.inject.Provider;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.pmiops.workbench.cdr.dao.ConceptDao;
+import org.pmiops.workbench.dataset.DataSetMapper;
+import org.pmiops.workbench.dataset.DataSetMapperImpl;
+import org.pmiops.workbench.db.dao.CdrVersionDao;
+import org.pmiops.workbench.db.dao.CohortDao;
+import org.pmiops.workbench.db.dao.ConceptSetDao;
+import org.pmiops.workbench.db.dao.DataDictionaryEntryDao;
+import org.pmiops.workbench.db.dao.DataSetDao;
+import org.pmiops.workbench.db.dao.DataSetService;
+import org.pmiops.workbench.db.model.CdrVersion;
+import org.pmiops.workbench.db.model.DataDictionaryEntry;
+import org.pmiops.workbench.db.model.User;
+import org.pmiops.workbench.firecloud.FireCloudService;
+import org.pmiops.workbench.model.Domain;
+import org.pmiops.workbench.notebooks.NotebooksService;
+import org.pmiops.workbench.test.FakeClock;
+import org.pmiops.workbench.workspaces.WorkspaceService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.liquibase.LiquibaseAutoConfiguration;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+import org.springframework.context.annotation.Scope;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+@DataJpaTest
+@Import(LiquibaseAutoConfiguration.class)
+public class DataDictionaryTest {
+
+  @Autowired BigQueryService bigQueryService;
+  @Autowired CohortDao cohortDao;
+  @Autowired ConceptDao conceptDao;
+  @Autowired ConceptSetDao conceptSetDao;
+  @Autowired DataSetDao dataSetDao;
+  @Autowired DataSetMapper dataSetMapper;
+  @Autowired DataSetService dataSetService;
+  @Autowired FireCloudService fireCloudService;
+  @Autowired NotebooksService notebooksService;
+  @Autowired WorkspaceService workspaceService;
+  @Mock Provider<User> userProvider;
+
+  @Autowired CdrVersionDao cdrVersionDao;
+  @Autowired DataDictionaryEntryDao dataDictionaryEntryDao;
+
+  private static final Instant NOW = Instant.now();
+  private static final FakeClock CLOCK = new FakeClock(NOW, ZoneId.systemDefault());
+  private static User currentUser;
+
+  @TestConfiguration
+  @Import({DataSetMapperImpl.class})
```

Comment:
I couldn't find this class in either PR. Is it generated somehow?

Reply:
Yep! Mapstruct generates this, fairly readable too. It's nice to be able to see what it produces for debugging.